pkg/asset/manifests/dns.go

```diff
@@ -115,7 +116,10 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			if err != nil {
 				return errors.Wrapf(err, "failed to get public zone for %q", installConfig.Config.BaseDomain)
 			}
-			config.Spec.PublicZone = &configv1.DNSZone{ID: zone.Name}
+			if installConfig.Config.GCP.PublicZoneID == "" {
```

Comment:
this makes an API call to get the zone, when it's already provided.. why not update this so that we only make the api call when PublicZoneID is no set.

Reply:
Good observation. I have reworked it to work as you stated.